### ISSUE TITLE
Creating Prospectors Log on 

### ISSUE THREAD

**Issue by Keridos** [GTNewHorizons/GT-New-Horizons-Modpack#18377]
### Your GTNH Discord Username

keridos

### Your Pack Version

2.7.0

### Your Server

private mp server

### Java Version

Java 21

### Type of Server

Vanilla Forge

### Your Expectation

Using the prospectors log (empty) should create a working copy of my veins. Other people should be able to use it to get my veins data.

### The Reality

I get kicked from the server due to a `java.lang.IllegalArgumentException: Payload may not be larger than 32k`. Server has no other info besides that my client lost the connection.
When logging in, I have a prospectors log that does not work for other people, even showing that I created it.

### Your Proposal

Fix the network code to properly handle the payload if its over 32kB and split it over multiple packets.
Maybe introduce some compression as I guess the data might be quite compressible.

### Final Checklist

- [X] I have searched this issue tracker and there is nothing similar already. Posting on a closed issue saying the bug still exists will prompt us to investigate and reopen it once we confirm your report.
- [X] I can reproduce this problem consistently by follow the exact steps I described above, or this does not need reproducing, e.g. recipe loophole.
- [ ] I have asked other people and they confirm they also have this problem by follow the exact steps I described above, or this does not need reproducing, e.g. recipe loophole.

**Comment by skyth540** [comment]
Me as well

**Comment by Hagvan** [comment]
Responding to bump as it is happening to me now as well.

**Comment by akdude2000** [comment]
Commenting here with a log, as I am having this issue as well.

having this issue in 2.7.4
Discord name is
akdude2000
private mp server
Java 21

vanilla forge for the server

Can not use a prospectors journal via right click

Joined server, right clicked with prospectors log and after DC and error, closed game and grabbed log
java.lang.IllegalArgumentException: Payload may not be larger than 32k
[fml-client-latest-issue.log](https://github.com/user-attachments/files/20936776/fml-client-latest-issue.log)

**Comment by Grishord** [comment]
We are having this issue as well in 2.7.4

**Comment by Ehony** [comment]
was using it fine when there was only a few veins on the map, when it gets more then i get crashed out from the server (bump)

**Comment by Keridos** [comment]
Yes, it is breaking at higher ore vein counts since there is too many to fit into a single packet.

If you have a private server you can use this to share data automatically between Serverutilities teams:
https://github.com/Lyfts/SharedProspecting

**Comment by TheBlueSquid** [comment]
I ran into this issue after only T3 planets and a fairly tame level of prospecting, it sucks not being able to use Prospector's logs anymore after you've done such a relatively small amount of prospecting. Will try the SharedProspecting mod!

**Comment by wlhlm** [comment]
I started looking into this, but am currently focused on helping with finalizing 2.8.

**Comment by Keridos** [comment]
> I started looking into this, but am currently focused on helping with finalizing 2.8.

Thanks.

I did not look into the source code but I think transferring the data from/to the server needs to be split into several chunks.
That should fix the issue.

**Comment by wlhlm** [comment]
The code already does that. It just miscalculates how many bytes the serialized prospecting data takes up.
